src/failure_reasons.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+enum class FailureReason {
+    REJECTED,
+    CANCELED,
+    BACKEND,
+    TIMEOUT,
```

Comment:
What's the difference or scenarios between REJECTED and TIMEOUT?

Reply:
Let's just go with rejected for now

Reply:
Keep rejected, remove timeout since timeout fits underneath rejected. There are multiple ways something can be rejected.

Reply:
Corrected.